app/content-panel/tags/page.tsx

```diff
@@ -1,15 +1,153 @@
-import AddTags from '../../../components/module-components/tags/AddTags';
 import TagTable from '../../../components/module-components/tags/TagTable';
+import TextInput from '../../../components/reusable-components/text-input/TextInput';
+import AddTag from '../../../components/module-components/tags/AddTag';
+import TagManagementControls from '../../../components/module-components/tags/TagManagementControls';
+import useDebounce from '../../../utils/hooks/useDebounce';
+import useGetTags from '../../../api/queries/tags/getTags';
+import useAddNewTag from '../../../api/mutations/tags/useAddNewTag';
+import useDeleteTag from '../../../api/mutations/tags/useDeleteTag';
+import useEditTag from '../../../api/mutations/tags/useEditTag';
+
+interface Tag {
+  id: string;
+  name: string;
+}
 
 const Tags = () => {
+  // MUTATIONS
+  const addNewTagMutation = useAddNewTag();
+  const deleteTagMutation = useDeleteTag();
+  const editTagMutation = useEditTag();
+
+  // STATE
+  const [showAddTag, setShowAddTag] = useState(false);
+  const [tags, setTags] = useState<Tag[]>([]);
+  const [editingTagId, setEditingTagId] = useState<string | null>(null);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const debouncedSearchTerm = useDebounce(searchTerm, 300);
+
+  const { data, isLoading } = useGetTags(debouncedSearchTerm);
+  const validationSchema = Yup.object().shape({
+    tagName: Yup.string()
+      .required('Името за тагот е задолжително')
+      .test('unique', 'Тагот веќе постои', function (value) {
+        return !tags.some((tag) => tag.name.toLowerCase() === value?.toLowerCase().trim());
+      }),
+  });
+
+  const handleDelete = async (id: string) => {
+    try {
+      await deleteTagMutation.mutateAsync(id);
+      setTags(tags.filter((tag) => tag.id !== id));
+      toast.success('Тагот беше успешно избришан.');
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const addTag = async (newTag: string) => {
+    const trimmedNewTag = newTag.trim();
+    if (tags.some((tag) => tag.name.toLowerCase() === trimmedNewTag.toLowerCase())) {
+      return { success: false, error: 'Тагот веќе постои.' };
+    }
+
+    try {
+      await addNewTagMutation.mutateAsync({
+        tagName: trimmedNewTag,
+      });
+
+      // If the mutation was successful, add the new tag to the local state
+      const newId = tags.length > 0 ? (parseInt(tags[tags.length - 1].id, 10) + 1).toString() : '1';
+      setTags([...tags, { id: newId, name: trimmedNewTag }]);
+      return { success: true };
+    } catch (error) {
+      // The error will be handled by the onError callback in useAddNewTag
+      return { success: false, error: 'API Error' };
+    }
+  };
+
+  const handleSaveChanges = (tagId: string, newName: string) => {
+    setTags((prevTags) =>
+      prevTags.map((tag) => (tag.id === tagId ? { ...tag, name: newName.trim() } : tag))
+    );
+    setEditingTagId(null);
+  };
```

Comment:
Lets refetch the data so we ensure that the data is always in sync with the server, and we won't have stale or inconsistent data. Here the dataset is relatively small so we should opt for refetching.

Reply:
in all crud operations here

Reply:
Done!